include/RevMem.h

```diff
@@ -246,15 +240,12 @@ class RevMem {
   // ----------------------------------------------------
   // ---- Atomic/Future/LRSC Interfaces
   // ----------------------------------------------------
-  /// RevMem: Add a memory reservation for the target address
-  bool LRBase( unsigned Hart, uint64_t Addr, size_t Len, void* Data, uint8_t aq, uint8_t rl, const MemReq& req, RevFlag flags );
-
-  /// RevMem: Clear a memory reservation for the target address
-  bool SCBase( unsigned Hart, uint64_t Addr, size_t Len, void* Data, void* Target, uint8_t aq, uint8_t rl, RevFlag flags );
-
   /// RevMem: Initiated an AMO request
   bool AMOMem( unsigned Hart, uint64_t Addr, size_t Len, void* Data, void* Target, const MemReq& req, RevFlag flags );
 
+  /// RevMem: Invalidate Matching LR reservations
+  bool InvalidateLRReservations( unsigned hart, uint64_t addr, size_t len );
```

Comment:
`InvalidateLRReservations()` cannot be `const` because it modifies `LRSC`.

`LRSC` is an `std::unordered_map` which maps between Hart ID and an (Addr, Len) pair, where the reservation is the half-open interval `[Addr, Addr+Len)`. Any other harts which write to this address range cause the reservation to be invalidated (erased). `SC` fails if no reservation from `LR` is found, or if not all of the `SC`'s address range lies in the reservation.

Reply:
Sorry should've specified I meant the arguments not the function

Reply:
https://stackoverflow.com/questions/117293/does-using-const-on-function-parameters-have-any-effect-why-does-it-not-affect

Reply:
Ahhh damn I thought I almost had ya :-)